src/lidar_subscriber.cpp

```diff
@@ -43,17 +48,17 @@ class LidarSubscriber : public rclcpp::Node
   private:
     rclcpp::Subscription<sensor_msgs::msg::LaserScan>::SharedPtr subscription;
     std::queue<Observation2D> laserscan_to_landmarks(const sensor_msgs::msg::LaserScan::SharedPtr msg){
-        std::vector<float> ranges = msg->ranges;
+        
         float angle_increment = msg->angle_increment;
         int measurement_count = static_cast<int>(std::round((msg->angle_max-msg->angle_min)/angle_increment)) + 1;
         std::queue<Observation2D> landmarks;
-        struct Observation2D first_measurement = {ranges[0],0};
+        struct Observation2D first_measurement = {msg->ranges[0],0};
         landmarks.push(first_measurement);
         std::queue<int> counts; 
         counts.push(1);
         for (int i = 1; i < measurement_count; i++){
-            float curr_range = ranges[i];
-            if (curr_range == std::numeric_limits<float>::infinity()){
+            float curr_range = msg->ranges[i];
+            if (curr_range == std::numeric_limits<float>::infinity()){ //This if statement could be eliminated if we get the laser_filters working
                 break;
```

Comment:
I think what you wanted to use here is `continue` instead of `break`, since `break` ends the loop, but `continue` continues on to the next iteration.